src/SALib/analyze/rsa.py

```diff
@@ -0,0 +1,283 @@
+        Unused, but defined to maintain compatibility.
+
+    References
+    ----------
+    1. Pianosi, F., K. Beven, J. Freer, J. W. Hall, J. Rougier, D. B. Stephenson, and
+    T. Wagener. 2016.
+    Sensitivity analysis of environmental models:
+    A systematic review with practical workflow.
+    Environmental Modelling & Software 79:214-232.
+    https://dx.doi.org/10.1016/j.envsoft.2016.02.008
+
+    2. Saltelli, A., M. Ratto, T. Andres, F. Campolongo, J. Cariboni, D. Gatelli,
+    M. Saisana, and S. Tarantola. 2008.
+    Global Sensitivity Analysis: The Primer.
+    Wiley, West Sussex, U.K.
+    https://dx.doi.org/10.1002/9780470725184
+    Accessible at:
+    http://www.andreasaltelli.eu/file/repository/Primer_Corrected_2022.pdf
+    """
+    groups = _check_groups(problem)
+    if not groups:
+        var_names = problem["names"]
+    else:
+        var_names, _ = extract_group_names(problem.get("groups", []))
+
+    results = rsa(X, Y, bins, target)
+
+    if groups:
+        groups = np.array(groups)
+        unique_grps = [*dict.fromkeys(groups)]
+        tmp = np.full((bins, len(unique_grps)), np.nan)
+
+        # Take the mean of effects from parameters that are grouped together
+        for grp_id, grp in enumerate(unique_grps):
+            tmp[:, grp_id] = np.nanmean(results[:, groups == grp], axis=1)
+
+        results = tmp
+        tmp = None
+
+    Si = ResultDict([(g, results[:, i]) for i, g in enumerate(var_names)])
+    Si["names"] = var_names
+    Si["bins"] = np.arange(0.0, 1.0, (1.0 / bins))
+    Si["target"] = target
+
+    # Attach object methods specific to this sensitivity method to ResultDict.
+    Si.to_df = MethodType(to_df, Si)
+    Si._plot = Si.plot
+    Si.plot = MethodType(plot, Si)
+
+    if print_to_console:
+        print(Si.to_df())
+
+    return Si
+
+
+def rsa(X: np.ndarray, y: np.ndarray, bins: int = 10, target="X") -> np.ndarray:
+    N, D = X.shape
+
+    # Pre-allocated arrays to store data/results
+    seq = np.append(np.arange(0.0, 1.0, (1 / bins)), 1.0)
+    X_di = np.zeros(N)
+    X_q = np.zeros(bins + 1)
+    r_s = np.full((bins, D), np.nan)
+    sel = np.zeros(N, dtype=bool)
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        for d_i in range(D):
+            X_di[:] = X[:, d_i]
+
+            if target == "X":
+                X_q[:] = np.quantile(X_di, seq)
+                sel[:] = (X_q[0] <= X_di) & (X_di <= X_q[1])
+            elif target == "Y":
+                y_q = np.quantile(y, seq)
+                sel[:] = (y_q[0] <= y) & (y <= y_q[1])
+
+            if (
+                (np.count_nonzero(sel) != 0)
+                and (len(y[~sel]) != 0)
+                and np.unique(y[sel]).size > 1
+            ):
+                if target == "X":
+                    r_s[0, d_i] = anderson_ksamp((y[sel], y[~sel])).statistic
+                elif target == "Y":
+                    r_s[0, d_i] = anderson_ksamp((X_di[sel], X_di[~sel])).statistic
```

Comment:
I am not sure I understand this correctly. I thought we should compare x for a given dimension with y?

Reply:
If the target is "X", then `sel` represents regions in factor space and we want to map those to `y`

If the target is "Y", then `sel` represents regions in output space, and we want to map those back to `X`.

Hope that's clearer, and I will add comments for clarity.